src/main/java/io/stargate/sgv2/jsonapi/service/updater/DocumentUpdater.java

```diff
@@ -70,6 +87,16 @@ private boolean update(ObjectNode docToUpdate, boolean docInserted) {
     return modified;
   }
 
+  /**
+   * vectorize UpdateClause as needed, only when documents are found or upsert will be used by
+   * updateOne, findOneAndUpdate, updateMany
+   *
+   * @param dataVectorizer
+   */
+  public Uni<Boolean> vectorizeUpdateClause(DataVectorizer dataVectorizer) {
+    return dataVectorizer.vectorizeUpdateClause(updateClause);
```

Comment:
Can we go through why the update clause is needed for the here, I would have though the previous and current versions of the document would be the best way to go